### PR TITLE
User approval feature

### PR DESCRIPTION
**Pull request by Romanitho**
# Proposed Changes

> 1st version of User Approval feature

image

image

## Related Issues

> ([Github link][autolink-references] to related issues or pull requests)

[autolink-references]: https://help.github.com/articles/autolinked-references-and-urls/

### CODE REVIEW

**Comment by KnifMelti** [comment]
The `$WAUConfig.WAU_UserApproval ` should/must be in **Get-Policies.ps1** and WAU-admx/adml too

**Comment by Romanitho** [comment]
And I'll probably change the txt file to an XML file with apps info... trying to get things faster a bit.

**Comment by Romanitho** [comment]
@KnifMelti a quick review to be sure ? not my part, so not sure if 'ive done it correctly 😅

**Comment by KnifMelti** [comment]
> @KnifMelti a quick review to be sure ? not my part, so not sure if I've done it correctly sweat_smile

When I have the time, yes

**Comment by Romanitho** [comment]
Yes. I will.

**Review by KnifMelti — COMMENTED** [review]
Tested out all OK!

Add in .admx.
Line 1:
`revision="4.?"` - (the revision you decide)
Line 11:
`<definition name="SUPPORTED_WAU_1_1?_?" displayName="$(string.SUPPORTED_WAU_1_1?_?)"/>`  - (the version you decide)
Line 356:
`<supportedOn ref="WAU:SUPPORTED_WAU_1_1?_?"/>`  - (the version you decide)

Add in .adml.
Line 1:
`revision="4.?"` - (the revision you decide)
Line 10:
`<string id="SUPPORTED_WAU_1_1?_?">Winget-AutoUpdate version 1.1?.? or later</string>`  - (the version you decide)

**Review by KnifMelti — APPROVED** [review]
Correct!
As an admx/adml Pro 👍

**Comment by Romanitho** [comment]
yes. I wanted to put 1.18.0 🤦🏻‍♂️
But you are my supervisor 👌🏻as always

File: Policies/WAU.admx [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <definitions>
       <definition name="SUPPORTED_WAU_1_16_0" displayName="$(string.SUPPORTED_WAU_1_16_0)"/>
       <definition name="SUPPORTED_WAU_1_16_5" displayName="$(string.SUPPORTED_WAU_1_16_5)"/>
+      <definition name="SUPPORTED_WAU_1_18_0" displayName="$(string.SUPPORTED_WAU_1_17_0)"/>
```
[the remark this answers is not in the document]
Comment (Romanitho) [reply Policies/WAU.admx:11]: oh yes, very tired ^^

**Comment by ivareri** [comment]
I'd like to see this feature implemented, is there anything I can do to help speed it up?

**Comment by Romanitho** [comment]
Sorry, just back from holidays 😊

You can do anything you want depending on your skills that could help to implement this feature 😇

**Pull request closed without merging**